indico_zodbimport/modules/event_papers.py

```diff
@@ -182,17 +306,78 @@ def _migrate_papers(self):
                 continue
             contribution = contributions[contrib_id]
             self._migrate_paper_roles(contribution)
+            self._migrate_revisions(contribution, rm)
+
+    def _migrate_paper_files(self):
+        n = 0
+        # we look for paper files in increasing order, so that we can detect file reuse
+        q = (LegacyPaperFile.find(Contribution.event_new == self.event).join(Contribution)
+                            .order_by(Contribution.id, LegacyPaperFile.revision_id))
+
+        for contribution_id, revision in itertools.groupby(q, attrgetter('contribution_id')):
+            for revision_id, files in itertools.groupby(revision, attrgetter('revision_id')):
+                ignored_checksums = set()
+                checksum_map = {}
+                if revision_id is not None:
+                    revision = self.legacy_contrib_revision_map[(contribution_id, revision_id)]
+                else:
+                    revision = None
+
+                for lpf in files:
+                    # check whether the same file has been uploaded to a subsequent revision
+                    try:
+                        with lpf.open() as f:
+                            checksum = crc32(f.read())
+                            checksum_map[checksum] = lpf
```

Comment:
I think you can already outdent here, you don't seem to be using `f` again below